File: src/main/java/nz/co/searchwellington/geocoding/osm/GeoCodeService.java
package nz.co.searchwellington.geocoding.osm;

import uk.co.eelpieconsulting.common.geo.model.OsmId;
import uk.co.eelpieconsulting.common.geo.model.Place;

public interface GeoCodeService {

	public Place resolveOsmId(OsmId osmId);
	
}

File: src/main/scala/nz/co/searchwellington/geocoding/osm/GeoCodeService.scala
package nz.co.searchwellington.geocoding.osm

import uk.co.eelpieconsulting.common.geo.model.{OsmId, Place}

import scala.concurrent.Future

trait GeoCodeService {
  def resolveOsmId(osmId: OsmId): Future[Option[Place]]
}